### PR TITLE
airbyte-ci: handle when user responds no to auto-upgrade prompt

### PR DESCRIPTION
**Pull request by erohmensing**
* Fix https://github.com/airbytehq/airbyte-internal-issues/issues/2774
* Don't error when the user responds `n`.
* Previously, I was going to have it exit 1 and say "can't run the command because the upgraded version is required. run it with `--disable-update-check` to bypass this". Seemed like a lot of steps when we already rerun the command after updating - we can rerun the command for them in the no case as well. I think this is a happy medium where we are not updating the user _for_ them, giving them the option to bypass if necessary, but still make it relatively easy to get around it if needed.

New behavior: 
```
~/airbytehq/airbyte/airbyte-ci/connectors/metadata_service/lib master *66 !1                                                                               6s  3.10 ─╮
❯ airbyte-ci-dev connectors --name=source-github list                                                                                                                ─╯
[23:27:58] INFO     root: Setting working directory to /Users/ella/airbytehq/airbyte                                                                    airbyte_ci.py:75
           WARNING  root:                                                                                                                             auto_update.py:109
                        🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨                                                                                  
                                                                                                                                                                        
                        The installed version of `airbyte-ci` does not match the version in your local airbyte repository.                                              
                        This likely means that your installed version of `airbyte-ci` is out of date.                                                                   
                                                                                                                                                                        
                        Installed Version: 2.10.4.                                                                                                                      
                        Local Repository Version: 2.10.7                                                                                                                
                                                                                                                                                                        
                        🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨                                                                                  
                                                                                                                                                                        
Do you want to automatically upgrade? [Y/n]: n
[23:28:00] INFO     root: Skipping auto-upgrade.                                                                                                      auto_update.py:132
           INFO     root: Re-running command with the `--disable-update-check` flag: /Users/ella/.local/bin/airbyte-ci-dev --disable-update-check     auto_update.py:133
                    connectors --name=source-github list                                                                                                                
[23:28:01] INFO     root: Setting working directory to /Users/ella/airbytehq/airbyte                                                                    airbyte_ci.py:75
           INFO     pipelines: GCP_GSM_CREDENTIALS environment variable found, using remote connector secrets.                                           commands.py:129
[23:28:02] INFO     pipelines: Will run on the following 1 connectors: source-github.                                                                     commands.py:25
           INFO     pipelines: Running Dagger Command list...                                                                              dagger_pipeline_command.py:34
           INFO     pipelines: If you're running this command for the first time the Dagger engine image will be pulled, it can take a     dagger_pipeline_command.py:35
                    short minute...                                                                                                                                     
           INFO     pipelines: Saving dagger logs to:                                                                                      dagger_pipeline_command.py:45
                    /Users/ella/airbytehq/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/list/manual/maste                              
                    r/1702466881/ce7554021f15fb981aacbf74031f9dfea5cf2143/dagger.log                                                                                    
                                              1 selected connectors                                              
┏━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Modified ┃ Connector     ┃ Language ┃ Release stage ┃ Version ┃ Folder                                        ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│          │ source-github │ python   │ certified     │ 1.5.4   │ airbyte-integrations/connectors/source-github │
└──────────┴───────────────┴──────────┴───────────────┴─────────┴───────────────────────────────────────────────┘
           INFO     pipelines: Dagger logs saved to                                                                                        dagger_pipeline_command.py:61
                    /Users/ella/airbytehq/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/list/manual/maste                              
                    r/1702466881/ce7554021f15fb981aacbf74031f9dfea5cf2143/dagger.log                                                                                    

~/airbytehq/airbyte/airbyte-ci/connectors/metadata_service/lib master *66 !1                                                                               6s  3.10 ─╮
❯ airbyte-ci-dev connectors --name=source-github list                                                                                                                ─╯
[23:28:53] INFO     root: Setting working directory to /Users/ella/airbytehq/airbyte                                                                    airbyte_ci.py:75
[23:28:54] WARNING  root:                                                                                                                             auto_update.py:109
                        🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨                                                                                  
                                                                                                                                                                        
                        The installed version of `airbyte-ci` does not match the version in your local airbyte repository.                                              
                        This likely means that your installed version of `airbyte-ci` is out of date.                                                                   
                                                                                                                                                                        
                        Installed Version: 2.10.4.                                                                                                                      
                        Local Repository Version: 2.10.7                                                                                                                
                                                                                                                                                                        
                        🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨                                                                                  
                                                                                                                                                                        
Do you want to automatically upgrade? [Y/n]: n
[23:29:06] INFO     root: Skipping auto-upgrade.                                                                                                      auto_update.py:132
           INFO     root: Re-running command with the `--disable-update-check` flag: /Users/ella/.local/bin/airbyte-ci-dev --disable-update-check     auto_update.py:133
                    connectors --name=source-github list                                                                                                                
[23:29:07] INFO     root: Setting working directory to /Users/ella/airbytehq/airbyte                                                                    airbyte_ci.py:75
[23:29:08] INFO     pipelines: GCP_GSM_CREDENTIALS environment variable found, using remote connector secrets.                                           commands.py:129
           INFO     pipelines: Will run on the following 1 connectors: source-github.                                                                     commands.py:25
           INFO     pipelines: Running Dagger Command list...                                                                              dagger_pipeline_command.py:34
           INFO     pipelines: If you're running this command for the first time the Dagger engine image will be pulled, it can take a     dagger_pipeline_command.py:35
                    short minute...                                                                                                                                     
           INFO     pipelines: Saving dagger logs to:                                                                                      dagger_pipeline_command.py:45
                    /Users/ella/airbytehq/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/list/manual/maste                              
                    r/1702466947/ce7554021f15fb981aacbf74031f9dfea5cf2143/dagger.log                                                                                    
                                              1 selected connectors                                              
┏━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Modified ┃ Connector     ┃ Language ┃ Release stage ┃ Version ┃ Folder                                        ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│          │ source-github │ python   │ certified     │ 1.5.4   │ airbyte-integrations/connectors/source-github │
└──────────┴───────────────┴──────────┴───────────────┴─────────┴───────────────────────────────────────────────┘
           INFO     pipelines: Dagger logs saved to                                                                                        dagger_pipeline_command.py:61
                    /Users/ella/airbytehq/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/list/manual/maste                              
                    r/1702466947/ce7554021f15fb981aacbf74031f9dfea5cf2143/dagger.log                                                                                    

~/airbytehq/airbyte/airbyte-ci/connectors/metadata_service/lib master *66 !1                                                                              17s  3.10 ─╮
❯ airbyte-ci-dev connectors --name=source-github list                                                                                                                ─╯
[23:29:16] INFO     root: Setting working directory to /Users/ella/airbytehq/airbyte                                                                    airbyte_ci.py:75
           WARNING  root:                                                                                                                             auto_update.py:109
                        🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨                                                                                  
                                                                                                                                                                        
                        The installed version of `airbyte-ci` does not match the version in your local airbyte repository.                                              
                        This likely means that your installed version of `airbyte-ci` is out of date.                                                                   
                                                                                                                                                                        
                        Installed Version: 2.10.4.                                                                                                                      
                        Local Repository Version: 2.10.7                                                                                                                
                                                                                                                                                                        
                        🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨                                                                                  
                                                                                                                                                                        
Do you want to automatically upgrade? [Y/n]: y
[23:29:18] INFO     root: [DEV] Upgrading pipelines...                                                                                                auto_update.py:117
[23:29:19] INFO     root: Setting working directory to /Users/ella/airbytehq/airbyte                                                                    airbyte_ci.py:75
           INFO     root: Updating to the latest development version of airbyte-ci...                                                                     commands.py:19
Python 3.10 is already installed.
pipx is already installed.
Installing to existing venv 'pipelines'
  installed package pipelines 2.10.7, installed using Python 3.10.10
  These apps are now globally available
    - airbyte-ci
    - airbyte-ci-dev
    - airbyte-ci-internal
done! ✨ 🌟 ✨
Development version of airbyte-ci installed.....
[23:29:29] INFO     root: Re-running command: /Users/ella/.local/bin/airbyte-ci-dev connectors --name=source-github list                              auto_update.py:123
[23:29:31] INFO     root: Setting working directory to /Users/ella/airbytehq/airbyte                                                                    airbyte_ci.py:75
           INFO     pipelines: airbyte-ci is up to date. Installed version: 2.10.7. Latest version: 2.10.7                                             auto_update.py:79
           INFO     pipelines: GCP_GSM_CREDENTIALS environment variable found, using remote connector secrets.                                           commands.py:129
[23:29:32] INFO     pipelines: Will run on the following 1 connectors: source-github.                                                                     commands.py:25
           INFO     pipelines: Running Dagger Command list...                                                                              dagger_pipeline_command.py:34
           INFO     pipelines: If you're running this command for the first time the Dagger engine image will be pulled, it can take a     dagger_pipeline_command.py:35
                    short minute...                                                                                                                                     
           INFO     pipelines: Saving dagger logs to:                                                                                      dagger_pipeline_command.py:45
                    /Users/ella/airbytehq/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/list/manual/maste                              
                    r/1702466971/ce7554021f15fb981aacbf74031f9dfea5cf2143/dagger.log                                                                                    
                                              1 selected connectors                                              
┏━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Modified ┃ Connector     ┃ Language ┃ Release stage ┃ Version ┃ Folder                                        ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│          │ source-github │ python   │ certified     │ 1.5.4   │ airbyte-integrations/connectors/source-github │
└──────────┴───────────────┴──────────┴───────────────┴─────────┴───────────────────────────────────────────────┘
           INFO     pipelines: Dagger logs saved to                                                                                        dagger_pipeline_command.py:61
                    /Users/ella/airbytehq/airbyte/airbyte-ci/connectors/pipelines/pipeline_reports/airbyte-ci/connectors/list/manual/maste                              
                    r/1702466971/ce7554021f15fb981aacbf74031f9dfea5cf2143/dagger.log            
```

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/pipelines/cli/auto_update.py [hunk 1]
```diff
@@ -98,36 +98,40 @@ def check_for_upgrade(
...
+    The installed version of `airbyte-ci` does not match the version in your local airbyte repository.
+    This likely means that your installed version of `airbyte-ci` is out of date.
 
     Installed Version: {__installed_version__}.
     Local Repository Version: {latest_version}
 
-    Please upgrade your local airbyte repository to the latest version using the following command:
-    $ {upgrade_command}
-
-    Alternatively you can skip this with the `--disable-update-check` flag.
```
Comment (erohmensing) [inline airbyte-ci/connectors/pipelines/pipelines/cli/auto_update.py:109]: the first doesn't seem necessary even before this pr, as we help them upgrade automatically now. the latter can be removed if we do the disabling for them as suggested in this PR. 

If we make it easy to bypass via the prompt, I think they're actually more likely to continue seeing the prompt (and therefore how out of date they are), than if appending `--disable-update-check` is the only way to successfully get the commands running on an old version. Then the flag becomes habit and some people are never prompted again

File: airbyte-ci/connectors/pipelines/pipelines/cli/auto_update.py [hunk 2]
```diff
@@ -98,36 +98,40 @@ def check_for_upgrade(
...
     """
     logging.warning(upgrade_error_message)
 
     # Ask the user if they want to upgrade
-    if enable_auto_update and confirm(
-        "Do you want to automatically upgrade?", default=True, additional_pre_confirm_key=AUTO_UPDATE_AGREE_KEY
-    ):
-        # if the current command contains `airbyte-ci-dev` is the dev version of the command
-        logging.info(f"[{'DEV' if is_dev_version else 'BINARY'}] Upgrading pipelines...")
+    if enable_auto_update:
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/cli/auto_update.py:112]: I'd find it more natural to prompt for upgrade `if enable_update_check and disable_auto_update`

[the remark this answers is not in the document]
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/cli/auto_update.py:112]: Agreed! I'm pretty sure that's what that variable actually means - can check and change it

[the remark this answers is not in the document]
Comment (erohmensing) [reply airbyte-ci/connectors/pipelines/pipelines/cli/auto_update.py:112]: Wait actually 

> I find it strange to ask for confirmation for an auto upgrade if enable_auto_update is already set to true...

Agreed! I'm pretty sure that's what that variable actually means - can check and change it

> I'd find it more natural to prompt for upgrade if enable_update_check and disable_auto_update

I'm not sure about that. It's the check that people are actively disabling, not the auto upgrade itself. Right? Let me take a look

**Pull request closed without merging**